### PR TITLE
Replaces a bunch of obj vars (and emagged on machinery/items) with obj_flags

### PR DESCRIPTION
**Pull request by vuonojenmustaturska**
Everything seems to work, but it feels a bit clunky to use. Maybe we should add bitfield helper defines?

oh, and i also removed force_blueprints on /obj as it's unused and unusable ???

:cl: Naksu
code: removed a bunch of unnecessary vars, combining them into a bitfield
/:cl:

TODO list for this PR:
- [x] Add bitfield helpers for dm, REMOVED: consensus seems to be that they're not needed
- [x] Figure out a way to help with the ergonomics on the mapping side

### CODE REVIEW

**Comment by optimumtact** [comment]
Excellent PR

File: _maps/RandomRuins/LavaRuins/lavaland_biodome_beach.dmm [hunk 1]
```diff
@@ -146,7 +146,7 @@
 /area/ruin/powered/beach)
 "aF" = (
 /obj/machinery/vending/boozeomat{
-	emagged = 1;
+	obj_flags = 5;
```
Comment (ChangelingRain) [inline _maps/RandomRuins/LavaRuins/lavaland_biodome_beach.dmm]: I greatly dislike this just from an 'ability to understand what the edit is without looking it up' standpoint.

Comment (vuonojenmustaturska) [reply _maps/RandomRuins/LavaRuins/lavaland_biodome_beach.dmm]: Would you be alright with /emagged subtypes? There's only a few unique types involved here

Comment (ninjanomnom) [reply _maps/RandomRuins/LavaRuins/lavaland_biodome_beach.dmm]: subtypes for the sake of mapping is just fine

Comment (ShizCalev) [reply _maps/RandomRuins/LavaRuins/lavaland_biodome_beach.dmm]: /obj/machinery/door/airlock/emagged
/obj/machinery/door/airlock/command/emagged
/obj/machinery/door/airlock/public/glass/emagged
/obj/machinery/door/airlock/research/emagged
/obj/machinery/vending/boozeomat/emagged
/obj/machinery/computer/arcade/battle/emagged
/obj/machinery/porta_turret/emagged
/mob/living/simple_animal/bot/secbot/beepsky/emagged

honestly this seems worse than just having an emagged = TRUE var tbh.

File: code/modules/awaymissions/mission_code/Cabin.dm [hunk 2]
```diff
@@ -72,7 +72,7 @@
 /obj/machinery/recycler/lumbermill
 	name = "lumbermill saw"
 	desc = "Faster then the cartoons!"
-	emagged = 2 //Always gibs people
+	obj_flags = CAN_BE_HIT | EMAGGED
```
Comment (ShizCalev) [inline code/modules/awaymissions/mission_code/Cabin.dm]: Changing behavior here.

File: _maps/RandomZLevels/research.dmm [hunk 3]
```diff
@@ -2729,7 +2729,7 @@
 "ij" = (
 /mob/living/simple_animal/bot/secbot/beepsky{
 	desc = "A beefy variant of the standard securitron model.";
-	emagged = 1;
```
Comment (ShizCalev) [inline _maps/RandomZLevels/research.dmm]: Emagged wasn't a boolean value for /mob/living/simple_animal/bot, just an FYI.

Comment (vuonojenmustaturska) [reply _maps/RandomZLevels/research.dmm]: oh, i shouldn't have removed this

File: code/modules/awaymissions/mission_code/Cabin.dm (hunk 2 above)
Comment (vuonojenmustaturska) [reply code/modules/awaymissions/mission_code/Cabin.dm]: Yeah, however I don't really care about behavior that, according to the comment on the code, only exists for admemes and is taken advantage of for an away mission that is never going to see the light of day on /tg/

File: _maps/RandomRuins/LavaRuins/lavaland_biodome_beach.dmm (hunk 1 above)
Comment (vuonojenmustaturska) [reply _maps/RandomRuins/LavaRuins/lavaland_biodome_beach.dmm]: beepsky isn't going to get one, changing it was an oversight

**Comment by monster860** [comment]
Can you please like.... not... with the defines?

**Comment by vuonojenmustaturska** [comment]
Regarding bitfield helpers, i wouldn't mind leaving them out, i only added them because someone asked. In general, however, no. If I can choose between having a proc and having a define, I will choose the define 10 times out of 10.

File: code/__DEFINES/obj_flags.dm [hunk 4]
```diff
@@ -0,0 +1,9 @@
+// Flags for the obj_flags var on /obj
+
+#define EMAGGED 1
+#define IN_USE 2 // If we have a user using us, this will be set on. We will check if the user has stopped using us, and thus stop updating and LAGGING EVERYTHING!
+#define CAN_BE_HIT 4 //can this be bludgeoned by items?
+#define BEING_SHOCKED 8 // Whether this thing is currently (already) being shocked by a tesla
+#define DANGEROUS_POSSESSION 16 //Admin possession yes/no
```
Comment (kevinz000) [inline code/__DEFINES/obj_flags.dm]: how do you plan to lock varedits without locking the entire flag entirely?
(and dont lock it entirely damnit)

Comment (vuonojenmustaturska) [reply code/__DEFINES/obj_flags.dm]: Why don't you look at /obj/vv_edit_var(vname, vval) in objs.dm to find out

**Comment by vuonojenmustaturska** [comment]
`var/set_obj_flags // ONLY FOR MAPPING: Sets flags from a string list, handled in Initialize. Usage: set_obj_flags = "EMAGGED;!CAN_BE_HIT" to set EMAGGED and clear CAN_BE_HIT.`

iirc @ShizCalev said this was good enough for them, what do other people think?

**Comment by optimumtact** [comment]
works for me

**Comment by vuonojenmustaturska** [comment]
I'll fix the conflicts when i have my PC in working order, the one time i take advantage of an internet deal it's for a part that decides to explode 10 months in just before weekend...

**Comment by Cyberboss** [comment]
@vuonojenmustaturska, sorry to rush you but the tracker is yuge

**Comment by vuonojenmustaturska** [comment]
let me just check real quick if i can add a bitfield helper to VV

**Comment by vuonojenmustaturska** [comment]
image

VV is now a bit more friendly, gonna need to add an edit interface too

**Comment by vuonojenmustaturska** [comment]
image
VV editing interface done, also a small refactor to tgalert-related code (it's now a descendant of a parent shared with the browser datum used for this list)

**Comment by vuonojenmustaturska** [comment]
@WJohn

**Comment by WJohn** [comment]
As far as mapping is concerned, I'd personally rather see it handled as subtypes, but I can understand that not being possible with how many airlock variations there are.

As long as you write this down someplace on how to make use of it, I'm sure it'll be fine. I rarely use emagged anyway.

**Comment by MoreRobustThanYou** [comment]
This looks good to me.

**Comment by AnturK** [comment]
Let's see how this goes.

**Pull request merged**